### DISCUSSION TITLE
Is it possible to have a seperator in Listview and no vertical grid alignment

### DISCUSSION THREAD

**Discussion by spiderbit** [davatorium/rofi#1839] [Q&A]
So I want basically ( item1 | item2 | item3 ) and if in line 1 the items are on average larger I see maybe 4 items there and then in second line 10 items), basically just like emacs shows it's horizontal dialogs.

**Comment by DaveDavenport** [comment]
I don't know what emacs does, but rofi listview shows for multi-line view  a grid  with fixed sized cells.

So currently not possible, if you want to add this, in the current design should not be to hard.

**Reply by spiderbit** [reply]
Well first there would be some separator support interesting sure you can put a "|" on the end of strings but that is only a workaround. Sure I get in a grid view the separator is maybe not as important but without it the non-grid view would also be problematic to use.

Would that also easy possible or is it already possible?

I also think vertical menus are superior in smaller / laptop screens but with big screens horizontal are better, and wasting space with this grid view is also negative.

I kind of tried to use some vertical aligned box in the middle for emacs on my netbook, but for my main or file dialogs as example it never could win me over. We have eyes that see a wider range of horizontal space than vertical. Also we can skim through books the idea with vertical / vertical alignment is that you have similar strings and can compare them somehow, but for many dialogs that is totally unnecessary. 

So you have 1 positive A box in the screen middle on one hand and better horizontal readability / skim-ability on the other hand, on big screens I think the latter is better.

**Reply by DaveDavenport** [reply]
I am confused trying to read this.
1.  For what is and isn't possible, please read documentation.
2.  I am unsure what you want, but you can customize a lot, including extra space (separation) between entries. (see 1)
3. ok. Then rofi might not be  the right tool for your task.
4. ?? I think this is not as universal as you make it out to be.  There are languages written top to bottom  rtl or ltr.    I prefer grid view as it is pretty deterministic where I need to read for each entry, dynamic entries take this away. I guess it is a personal preference. 
5. ?? what is a positive A box?

**Comment by spiderbit** [comment]
I think that is kind of related, there seems to be a "barview" code that as far as I understand should be even in upstream included. That does not really solve the seperator question and it might be limidet to 1 line view only, but it's better than nothing or a start I guess.
#252

**Reply by DaveDavenport** [reply]
yes, hence the 'multi-line'  in the reply.
This code is not very great and has it own issues.

**Comment by spiderbit** [comment]
Well the dmenu option seem to ignore my font parameter and also seem to be fixed limited to 1 line. But would probably the starting point to make that happen. It's interesting that old dmenu / devs agrees with me that horizontal non-grid layouts are a good idea :D

**Reply by DaveDavenport** [reply]
What is 'font parameter'

**Reply by DaveDavenport** [reply]
>  It's interesting that old dmenu / devs agrees with me that horizontal non-grid layouts are a good idea :D

Then use dmenu.
